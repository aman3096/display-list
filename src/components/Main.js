
import React from "react";
import { useQuery, gql } from "@apollo/client";
import { Link } from "react-router-dom";
export const FILMS_QUERY = gql`
{
    allFilms {
      films {
        title
        director
        releaseDate
        speciesConnection {
          species {
            name
            classification
            homeworld {
              name
            }
          }
        }
      }
  }
}
`;

function Main(){
    const { data, loading, error } = useQuery(FILMS_QUERY);
    if (loading) return "Loading...";
    if (error) return <pre>{error.message}</pre>
    return (
      <div className="container" >
        <center data-testid="main-2"><h1>Star Wars All Films</h1></center>
        <ul className="list-group list-group-flush">
          {data.allFilms.films.map((film,index)=><li className={index%2===1?"list-group-item list-group-item-dark":"list-group-item list-group-item-light"} key={"film-"+film.title}><Link to="/details" state={{dats:film}}>{film.title}</Link></li>)}
        </ul>
      </div>
    );
}

export default Main;