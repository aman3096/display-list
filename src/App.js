import React from "react";
// import { useQuery, gql } from "@apollo/client";
import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
import Detail from "./components/Detail";
import Main from "./components/Main";
import { ApolloProvider, ApolloClient, InMemoryCache } from "@apollo/client";


const client = new ApolloClient({
  uri: "https://swapi-graphql.netlify.app/.netlify/functions/index",
  cache: new InMemoryCache(),
  credentials:'omit'
});

export default function App() {
  return (
    <ApolloProvider client={client}>
    <div className="container" data-testid="main-1">
      <Router>
        <Routes>
          <Route path="/" index element={<Main/>}>
          </Route>
          <Route path="/details" element={<Detail />} >
          
          </Route>
          </Routes>
      </Router>
    </div>
    </ApolloProvider>
  );
}