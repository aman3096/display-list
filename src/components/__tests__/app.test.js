import { render, screen, cleanup } from "@testing-library/react"
import renderer from "react-test-renderer";
import Main from "../Main";
import App from "../../App";

afterEach(()=> {
   cleanup();
});

test('should render App component',() => {
   render(<App/>);
   const MainElement = screen.getByTestId('main-1');
   expect(MainElement).toBeInTheDocument();
   expect(MainElement).toHaveTextContent('Loading...');
})

test('should not contain strike component', ()=>{
   render(<App/>);
   const Main = screen.getByTestId('main-1');
   expect(Main).not.toContainHTML('strike');
})

test('matches snapshot', ()=> {
   const tree = renderer.create(<App/>).toJSON();
   console.log(tree);
   expect(tree).toMatchSnapshot();
})