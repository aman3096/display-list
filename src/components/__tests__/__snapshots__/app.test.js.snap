// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="container"
  data-testid="main-1"
>
  Loading...
</div>
`;
